File: examples/java-maven-example/src/test/java/org/galaxio/performance/picatinny/Stability.java
package org.galaxio.performance.picatinny;

import io.gatling.javaapi.core.OpenInjectionStep;
import org.galaxio.gatling.javaapi.SimulationConfig;
import org.galaxio.gatling.javaapi.SimulationWithTransactions;
import org.galaxio.gatling.javaapi.Utility;
import org.galaxio.performance.picatinny.scenarios.PicatinnyScenario;

import static io.gatling.javaapi.core.CoreDsl.*;

public final class Stability extends SimulationWithTransactions {
    {
        OpenInjectionStep[] injectionProfile = {
            rampUsersPerSec(0).to(SimulationConfig.intensity()).during(SimulationConfig.rampDuration()),
            constantUsersPerSec(SimulationConfig.intensity()).during(SimulationConfig.stageDuration()),
        };

        Utility.banner(injectionProfile);
        Utility.diagnostics();

        setUp(
            PicatinnyScenario.apply("Picatinny Stability", "java-stability")
                .injectOpen(injectionProfile)
        ).maxDuration(PerformanceSupport.toScala(SimulationConfig.testDuration()))
            .assertions(global().failedRequests().count().is(0L));
    }
}

File: examples/kotlin-gradle-example/src/gatling/kotlin/org/galaxio/performance/picatinny/Stability.kt
package org.galaxio.performance.picatinny

import io.gatling.javaapi.core.CoreDsl.constantUsersPerSec
import io.gatling.javaapi.core.CoreDsl.rampUsersPerSec
import io.gatling.javaapi.core.OpenInjectionStep
import org.galaxio.gatling.javaapi.SimulationConfig
import org.galaxio.gatling.javaapi.Utility
import org.galaxio.gatling.javaapi.SimulationWithTransactions
import org.galaxio.performance.picatinny.scenarios.PicatinnyScenario

class Stability : SimulationWithTransactions() {
    init {
        val injectionProfile = arrayOf<OpenInjectionStep>(
            rampUsersPerSec(0.0).to(SimulationConfig.intensity()).during(SimulationConfig.rampDuration()),
            constantUsersPerSec(SimulationConfig.intensity()).during(SimulationConfig.stageDuration()),
        )

        Utility.banner(*injectionProfile)
        Utility.diagnostics()

        setUp(
            PicatinnyScenario.apply("Picatinny Stability", "kotlin-stability")
                .injectOpen(*injectionProfile),
        ).maxDuration(PerformanceSupport.toScala(SimulationConfig.testDuration()))
            .assertions(PerformanceSupport.noFailedRequests())
    }
}
